ckan/config/middleware/__init__.py

```diff
@@ -151,8 +151,6 @@ def handle_i18n(self, environ):
             qs = environ.get('QUERY_STRING')
 
             if qs:
-                # sort out weird encodings
-                qs = urllib.quote(qs, '')
```

Comment:
this is done above in line 149 as well. does it make sense to remove only this instance?

Reply:
I don't know. The path info part didn't trigger my error, but my test cases may not have hit it. It doesn't seem to be required from a quick attempt at poking at it. 

This particular issue with this code is that `urllib.quote(qs)` encodes the `&` and `=` in the querystring, and then created an invalid url that required further processing. I don't think that's actually the case with the pathinfo, as it's splitting on the separators already.